src/k2/module/k23si/Module.cpp

```diff
@@ -499,15 +506,16 @@ Status K23SIPartitionModule::_validateStaleWrite(const RequestT& request, Versio
     // if a txn committed a value at time T5, then we must also assume they did a read at time T5
     // NB(3) if we encounter a WI, we check the second oldest version to see if there is a need to push.
     // If the second oldest is newer than we are, then we won't commit even if we win a PUSH against the WI.
```

Comment:
Should we consider the op ID now for stale write? I.e. lower sequence# should be considered stale against a higher one